draft-rescorla-tls-ctls.md

```diff
@@ -506,6 +511,46 @@ Total: 113 + X bytes
 [TODO]
 
 
+# cTLS as Compression Layer [[OPEN ISSUE]]
+
+The above text treates cTLS as a new protocol; however it is also possible
+to view it as a form of compression for TLS, which sits in between the
+handshake layer and the record layer, like so:
+
+~~~~~
++---------------+---------------+---------------+
+|   Handshake   |  Application  |     Alert     |
++---------------+---------------+---------------+
+|               cTLS Compression Layer          |
++---------------+---------------+---------------+
+|               cTLS Record Layer               |
++---------------+---------------+---------------+
+~~~~~
+
+This structure does involve one technical difference: because the handshake
+message transformation happens below the handshake layer, the cTLS handshake
+transcript would be the same as the TLS 1.3 handshake transcript. This has
+both advantages and disadvantages.
+
+The major advantage is that it makes it possible to reuse all the TLS
+security proofs even with very aggressive compression (with suitable
+proofs about the bijectiveness of the compression). [Thanks to Karthik
+Bhargavan for this point.] This probably also makes it easier to
+implement more aggressive compression. For instance, the above text
+shrinks the handshake headers but does not elide them entirely.
+If the handshake shape (i.e., which messages are sent) is known in
+advance, then these headers can be removed, thus trimming about 20 bytes
+from the handshake. This is easier to reason about as a form of compression.
+With very aggressive parameters, this technique can bring the
+handshake down to about XXX [[RLB]] bytes.
```

Comment:
XXX = 48 / 164 / 116 for M1 / M2 / M3, with:
* Mutual auth
* 16 byte random
* Finished included